sw/device/lib/crypto/drivers/kmac.c

```diff
@@ -656,3 +665,107 @@ kmac_error_t kmac_process_msg_blocks(kmac_operation_t operation,
 
   return kKmacOk;
 }
+
+OT_WARN_UNUSED_RESULT
+kmac_error_t kmac_sha3_224(const uint8_t *data, size_t data_len,
+                           uint8_t *digest) {
+  if (data == NULL && data_len > 0) {
+    return kKmacArgsError;
+  } else if (digest == NULL) {
+    return kKmacArgsError;
+  }
```

Comment:
I think these checks can be removed because dereferencing a NULL ptr will most likely shutdown the chip -- I assume we will have a handler that triggers a shutdown.

Reply:
I'd be ok with that but I'm not sure if there are any prior discussions/decisions about that. @moidx @jadephilipoom